app/components/modals/add-system-role-modal.js

```diff
@@ -10,6 +10,12 @@ export default ModalBase.extend(FormMixin, {
       this.onValid(() => {
         this.sendAction('addSystemRole');
       });
+    },
+    close() {
+      if (!this.get('role').get('id')) {
```

Comment:
umm..does `this.get('role.id')` not work ?

Reply:
Yes, it works. Thanks for pointing out, I have made the changes. Please review :)